src/data/nat/choose.lean

```diff
@@ -9,13 +9,27 @@ open nat
 
 open_locale big_operators
 
-lemma nat.prime.dvd_choose {p k : ℕ} (hk : 0 < k) (hkp : k < p) (hp : prime p) : p ∣ choose p k :=
-have h₁ : p ∣ fact p, from hp.dvd_fact.2 (le_refl _),
-have h₂ : ¬p ∣ fact k, from mt hp.dvd_fact.1 (not_le_of_gt hkp),
-have h₃ : ¬p ∣ fact (p - k), from mt hp.dvd_fact.1 (not_le_of_gt (nat.sub_lt_self hp.pos hk)),
-by rw [← choose_mul_fact_mul_fact (le_of_lt hkp), mul_assoc, hp.dvd_mul, hp.dvd_mul] at h₁;
+lemma nat.prime.dvd_choose_add {p a b : ℕ} (hap : a < p) (hbp : b < p) (h : p ≤ a + b)
+  (hp : prime p) : p ∣ choose (a + b) a :=
+have h₁ : p ∣ fact (a + b), from hp.dvd_fact.2 h,
+have h₂ : ¬p ∣ fact a, from mt hp.dvd_fact.1 (not_le_of_gt hap),
+have h₃ : ¬p ∣ fact b, from mt hp.dvd_fact.1 (not_le_of_gt hbp),
+by
+  rw [← choose_mul_fact_mul_fact (le.intro rfl), mul_assoc, hp.dvd_mul, hp.dvd_mul,
+    norm_num.sub_nat_pos (a + b) a b rfl] at h₁;
   exact h₁.resolve_right (not_or_distrib.2 ⟨h₂, h₃⟩)
 
+lemma nat.prime.dvd_choose_self {p k : ℕ} (hk : 0 < k) (hkp : k < p) (hp : prime p) :
+  p ∣ choose p k :=
+begin
+  have r : k + (p - k) = p,
+    by rw [← nat.add_sub_assoc (nat.le_of_lt hkp) k, nat.add_sub_cancel_left],
+  have e : p ∣ choose (k + (p - k)) k,
+    by exact nat.prime.dvd_choose_add hkp (sub_lt (lt.trans hk hkp) hk) (by rw r) hp,
+  rw r at e,
+  exact e,
```

Comment:
```suggestion
  rwa r at e,
```
`rwa` does the same thing as `rw ...` then `assumption` (and `assumption` is the same as `exact h` for any applicable `h` in the context).